From: Douglas Anderson <dianders@chromium.org>
Subject: init-mainc-wrap-long-kernel-cmdline-when-printing-to-logs-v4
Date: Mon, 27 Oct 2025 08:22:13 -0700

- Use 0 instead of -1 to disable wrapping.
- If COMMAND_LINE_SIZE makes wrapping impossible then help compiler optimize.

Link: https://lkml.kernel.org/r/20251027082204.v4.1.I095f1e2c6c27f9f4de0b4841f725f356c643a13f@changeid
Signed-off-by: Douglas Anderson <dianders@chromium.org>
Tested-by: Paul E. McKenney <paulmck@kernel.org>
Cc: Alexander Shishkin <alexander.shishkin@linux.intel.com>
Cc: Andrew Chant <achant@google.com>
Cc: Brian Gerst <brgerst@gmail.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: Francesco Valla <francesco@valla.it>
Cc: Geert Uytterhoeven <geert@linux-m68k.org>
Cc: guoweikang <guoweikang.kernel@gmail.com>
Cc: Heiko Carstens <hca@linux.ibm.com>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: Ingo Molnar <mingo@kernel.org>
Cc: Jan Hendrik Farr <kernel@jfarr.cc>
Cc: Jeff Xu <jeffxu@chromium.org>
Cc: Kees Cook <kees@kernel.org>
Cc: Masahiro Yamada <masahiroy@kernel.org>
Cc: Michal Koutný <mkoutny@suse.com>
Cc: Miguel Ojeda <ojeda@kernel.org>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Nathan Chancellor <nathan@kernel.org>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Randy Dunlap <rdunlap@infradead.org>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Cc: Sven Schnelle <svens@linux.ibm.com>
Cc: Tejun Heo <tj@kernel.org>
Cc: Thomas Gleinxer <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/Kconfig |    4 ++--
 init/main.c  |    5 +++--
 2 files changed, 5 insertions(+), 4 deletions(-)

--- a/init/Kconfig~init-mainc-wrap-long-kernel-cmdline-when-printing-to-logs-v4
+++ a/init/Kconfig
@@ -1515,7 +1515,7 @@ config BOOT_CONFIG_EMBED_FILE
 config CMDLINE_LOG_WRAP_IDEAL_LEN
 	int "Length to try to wrap the cmdline when logged at boot"
 	default 1021
-	range -1 1021
+	range 0 1021
 	help
 	  At boot time, the kernel command line is logged to the console.
 	  The log message will start with the prefix "Kernel command line: ".
@@ -1526,7 +1526,7 @@ config CMDLINE_LOG_WRAP_IDEAL_LEN
 	  ideal length if a place to wrap isn't found before the specified
 	  number of characters.
 
-	  A value of -1 disables wrapping, though be warned that the maximum
+	  A value of 0 disables wrapping, though be warned that the maximum
 	  length of a log message (1021 characters) may cause the cmdline to
 	  be truncated.
 
--- a/init/main.c~init-mainc-wrap-long-kernel-cmdline-when-printing-to-logs-v4
+++ a/init/main.c
@@ -948,8 +948,9 @@ static void __init print_kernel_cmdline(
 {
 	size_t len;
 
-	/* Config option of -1 disables wrapping */
-	if (CONFIG_CMDLINE_LOG_WRAP_IDEAL_LEN < 0) {
+	/* Config option of 0 or anything longer than the max disables wrapping */
+	if (CONFIG_CMDLINE_LOG_WRAP_IDEAL_LEN == 0 ||
+	    IDEAL_CMDLINE_LEN >= COMMAND_LINE_SIZE - 1) {
 		pr_notice("%s%s\n", KERNEL_CMDLINE_PREFIX, cmdline);
 		return;
 	}
_
